pkg/kademlia/lookup.go

```diff
@@ -5,24 +5,30 @@ import (
 	"context"
 	"log"
 	"math/big"
+	"sync"
 	"time"
 
 	"storj.io/storj/pkg/dht"
 	"storj.io/storj/pkg/node"
 	"storj.io/storj/pkg/pb"
 )
 
-type sequentialLookup struct {
+var (
+	defaultRoutines = 5
+)
+
+type lookup struct {
 	contacted       map[string]bool
+	lMutex          *sync.Mutex
 	queue           PriorityQueue
 	slowestResponse time.Duration
 	client          node.Client
 	target          dht.NodeID
-	limit           int
-	bootstrap       bool
+	routines        int
+	rMutex          *sync.Mutex
```

Comment:
It's unclear where these lMutex and rMutex are exactly used. Also they don't need to be pointers.

Usually the code is organized like:

```
type discovery struct {
    client node.Client
    target dht.NodeID
    routines int

    mu        sync.Mutex
    nowork    sync.Cond
    queue     PriorityQueue
    contacted map[string]bool

    muSlowest       sync.Mutex
    slowestResponse time.Duration
}
```

This way it can be seen mu is protecting the following structures and muSlowest the slowestResponse. Similarly it can be seen that client and target can be used without locking.